bokehjs/src/coffee/model.coffee

```diff
@@ -17,6 +21,42 @@ export class Model extends HasProps
       for cb in callbacks
         @listenTo(@, evt, () -> cb.execute(@))
 
+    @listenTo(@, 'change:js_event_callbacks', () -> @_update_event_callbacks)
+
+
+  _process_event : (event) ->
+    # Given an applicable event, execute the associated callback
+    if not event.constructor.event_name
+       logger.warn('Event is still passed as a string: ' + event)
+       return
+    if not event.constructor.applicable_models.some((m) => m == @.type)
```

Comment:
@mattpap I had a go at this but didn't have any luck. I don't get why the ``instanceof`` check never passes in the following diff:

```diff
modified   bokehjs/src/coffee/core/bokeh_events.ts
@@ -1,4 +1,6 @@
 import { logger } from "./logging"
+import {Button} from "../models/widgets"
+import {Plot} from "../models/plots/plot"
 
 export class BokehEvent {
   static _event_classes: any = {}
@@ -54,14 +56,14 @@ export class BokehEvent {
 
 export class ButtonClick extends BokehEvent {
   static event_name = 'button_click'
-  static applicable_models = ['Button']
+  static applicable_models = [Button]
 }
 
 
 export class UIEvent extends BokehEvent {
   // A UIEvent is an event originating on a PlotCanvas this includes
   // DOM events such as keystrokes as well as hammer events and LOD events.
-  static applicable_models = ['Plot']
+  static applicable_models = [Plot]
 }
 
 
modified   bokehjs/src/coffee/model.coffee
@@ -29,7 +29,8 @@ export class Model extends HasProps
     if not event.constructor.event_name
        logger.warn('Event is still passed as a string: ' + event)
        return
-    if not event.constructor.applicable_models.some((m) => m == @.type)
+
+    if not event.constructor.applicable_models.some((m) => m instanceof @.constructor)
       return
 
     event = event._customize_event(@)
```